packages/instantsearch.js/src/widgets/chat/chat.tsx

```diff
@@ -274,6 +276,159 @@ function createDefaultTools<
   };
 }
 
+type ChatWrapperProps = {
+  cssClasses: ChatCSSClasses;
+  chatOpen: boolean;
+  setChatOpen: (open: boolean) => void;
+  chatMessages: ChatMessageBase[];
+  setChatMessages: (messages: ChatMessageBase[]) => void;
+  indexUiState: IndexUiState;
+  setIndexUiState: IndexWidget['setIndexUiState'];
+  chatStatus: ChatStatus;
+  chatInput: ChatRenderState['input'];
+  setChatInput: ChatRenderState['setInput'];
+  sendMessage: ChatRenderState['sendMessage'];
+  regenerate: ChatRenderState['regenerate'];
+  stop: ChatRenderState['stop'];
+  toolsForUi: ClientSideTools;
+  toggleButtonProps: {
+    layoutComponent: ComponentProps<typeof Chat>['toggleButtonComponent'];
+    iconComponent: ComponentProps<
+      typeof Chat
+    >['toggleButtonProps']['toggleIconComponent'];
+  };
+  headerProps: {
+    layoutComponent: ComponentProps<typeof Chat>['headerComponent'];
+    closeIconComponent: ChatHeaderProps['closeIconComponent'];
+    minimizeIconComponent: ChatHeaderProps['minimizeIconComponent'];
+    maximizeIconComponent: ChatHeaderProps['maximizeIconComponent'];
+    titleIconComponent: ChatHeaderProps['titleIconComponent'];
+    translations: Partial<ChatHeaderTranslations>;
+  };
+  messagesProps: {
+    loaderComponent:
+      | ((props: ChatMessageLoaderProps) => JSX.Element)
+      | undefined;
+    errorComponent: ((props: ChatMessageErrorProps) => JSX.Element) | undefined;
+    actionsComponent:
+      | ((props: { actions: ChatMessageActionProps[] }) => JSX.Element)
+      | undefined;
+    translations: Partial<ChatMessagesTranslations>;
+  };
+  promptProps: {
+    layoutComponent: ComponentProps<typeof Chat>['promptComponent'];
+    headerComponent: ChatPromptProps['headerComponent'];
+    footerComponent: ChatPromptProps['footerComponent'];
+    translations: Partial<ChatPromptTranslations>;
+    promptRef: { current: HTMLTextAreaElement | null };
+  };
+  state: ReturnType<typeof createLocalState>;
+};
+
+function ChatWrapper({
```

Comment:
Can this not be done inline in the `render` fn?

Reply:
I tried, but the whole chat re-renders when I type something into the chat prompt